soh/soh/Enhancements/randomizer/location_access/dungeons/fire_temple.cpp

```diff
@@ -506,114 +631,174 @@ void RegionTable_Init_FireTemple() {
         Entrance(RR_FIRE_TEMPLE_MQ_BIG_LAVA_ROOM, []{return true;}),
     });
 
-    //This room assumes Goron Tunic until looser tunic requirements tricks are made
-    areaTable[RR_FIRE_TEMPLE_MQ_ELEVATOR_ROOM] = Region("Fire Temple MQ Elevator Room", SCENE_FIRE_TEMPLE, {}, {
+    areaTable[RR_FIRE_TEMPLE_MQ_LAVA_GEYSER_1F] = Region("Fire Temple MQ Lava Geyser 1F", SCENE_FIRE_TEMPLE, {}, {
         //Locations
         LOCATION(RC_FIRE_TEMPLE_MQ_FIRE_PILLAR_LEFT_HEART,  true),
         LOCATION(RC_FIRE_TEMPLE_MQ_FIRE_PILLAR_RIGHT_HEART, true),
         LOCATION(RC_FIRE_TEMPLE_MQ_FIRE_PILLAR_LOWER_HEART, true),
     }, {
         //Exits
-        Entrance(RR_FIRE_TEMPLE_MQ_BIG_LAVA_ROOM,  []{return true;}),
-        Entrance(RR_FIRE_TEMPLE_MQ_BIG_TORCH_ROOM, []{return true;}),
+        Entrance(RR_FIRE_TEMPLE_MQ_BIG_LAVA_ROOM,       []{return true;}),
+        Entrance(RR_FIRE_TEMPLE_MQ_LAVA_GEYSER_GRATE,   []{return logic->FireTimer() >= 40/* && CanClimbHigh()*/;}),
+        //this technically only reaches torch pillar, but the heart pillar can be reached from there with longshot
+        Entrance(RR_FIRE_TEMPLE_MQ_LAVA_GEYSER_PILLARS, []{return logic->FireTimer() >= 40 && logic->CanUse(RG_LONGSHOT);}),
+    });
+    
+    //tunic logic handled on entry
+    areaTable[RR_FIRE_TEMPLE_MQ_LAVA_GEYSER_GRATE] = Region("Fire Temple MQ Lava Geyser Grate", SCENE_FIRE_TEMPLE, {}, {
+        //Locations
+        LOCATION(RC_FIRE_TEMPLE_MQ_FIRE_PILLAR_LEFT_HEART,  true),
+        LOCATION(RC_FIRE_TEMPLE_MQ_FIRE_PILLAR_RIGHT_HEART, true),
+    }, {
+        //Exits
+        Entrance(RR_FIRE_TEMPLE_MQ_LAVA_GEYSER_1F,      []{return true;}),
+        Entrance(RR_FIRE_TEMPLE_MQ_LAVA_GEYSER_PILLARS, []{return true;}),
     });
 
-    areaTable[RR_FIRE_TEMPLE_MQ_BIG_TORCH_ROOM] = Region("Fire Temple MQ Big Torch Room", SCENE_FIRE_TEMPLE, {}, {}, {
+    //tunic logic handled on entry
+    areaTable[RR_FIRE_TEMPLE_MQ_LAVA_GEYSER_PILLARS] = Region("Fire Temple MQ Lava Geyser Pillars", SCENE_FIRE_TEMPLE, {}, {
+        //Locations
+        //rang can get the other 2 hearts from here but it's an awkward shot so a trick
+        LOCATION(RC_FIRE_TEMPLE_MQ_FIRE_PILLAR_LOWER_HEART, true),
+    }, {
         //Exits
-        Entrance(RR_FIRE_TEMPLE_MQ_LOWER_MAZE,         []{return (logic->HasFireSource() && logic->CanUse(RG_HOOKSHOT)) || (ctx->GetTrickOption(RT_FIRE_MQ_CLIMB) && logic->CanUse(RG_HOVER_BOOTS));}),
-        Entrance(RR_FIRE_TEMPLE_MQ_ELEVATOR_ROOM,      []{return logic->CanUse(RG_GORON_TUNIC);}),
-        Entrance(RR_FIRE_TEMPLE_MQ_MAZE_SHORTCUT_CAGE, []{return logic->Get(LOGIC_FIRE_OPENED_UPPER_SHORTCUT);}),
+        Entrance(RR_FIRE_TEMPLE_MQ_LAVA_GEYSER_1F,    []{return true;}),
+        Entrance(RR_FIRE_TEMPLE_MQ_LAVA_GEYSER_GRATE, []{return logic->CanUse(RG_HOOKSHOT);}),
+        Entrance(RR_FIRE_TEMPLE_MQ_LAVA_GEYSER_2F,    []{return true;}),
```

Comment:
```suggestion
        Entrance(RR_FIRE_TEMPLE_MQ_LAVA_GEYSER_2F,    []{return true /*str0*/;}),
```
tho I think this needs to be event for doorsanity as going to first floor without first pushing block takes damage

Reply:
This is MQ, the block starts pre-pushed in MQ, and you can escape the block's immobilization by rolling and thus skip the fall/fire damage when riding down. I don't consider this a trick as I consider all OHKO specific logic a trick.